src/stores/safariKeyboardStore.ts

```diff
@@ -0,0 +1,38 @@
+import { isIOS, isSafari, isTouch } from '../browser'
+import * as selection from '../device/selection'
+import reactMinistore from './react-ministore'
+
+const VIRTUAL_KEYBOARD_CLOSE_DURATION = 800
+
+const safariKeyboardStore = reactMinistore<{
+  open: boolean
+  closing: boolean
+}>({
+  open: false,
+  closing: false,
+})
+
+/** Updates the safariKeyboardStore state based on the selection. */
+const updateKeyboardState = () => {
+  const keyboardIsVisible = selection.isActive()
+  if (safariKeyboardStore.getState().open && !keyboardIsVisible) {
+    safariKeyboardStore.update({
+      closing: true,
+    })
+    setTimeout(() => {
+      safariKeyboardStore.update({
+        closing: false,
+      })
+    }, VIRTUAL_KEYBOARD_CLOSE_DURATION)
+  }
+  safariKeyboardStore.update({
+    open: keyboardIsVisible,
+  })
+}
+
+if (isTouch && isSafari() && !isIOS) {
+  updateKeyboardState()
+  document.addEventListener('selectionchange', updateKeyboardState)
```

Comment:
I prefer to initialize document and events in a dedicated loading phase rather than in the module scope. It creates more predictable timing, and allows event listeners to be removed in a cleanup phase if needed.

Therefore, I recommend exporting `updateKeyboardState` and calling it in the existing global `onSelectionChange`:

https://github.com/cybersemics/em/blob/71545c5dbe3eda5204cf00a02590cafc48b2f58d/src/util/initEvents.ts#L164-L171